tests/test_env.py

```diff
@@ -1,20 +1,101 @@
 from __future__ import annotations
 
-from pathlib import Path
+from dataclasses import dataclass
+from typing import Dict
 
 import pytest
 
 
-@pytest.fixture()
-def example(testdir: pytest.Testdir) -> pytest.Testdir:
-    src = Path(__file__).parent / "example.py"
-    dest = Path(str(testdir.tmpdir / "test_example.py"))
-    dest.symlink_to(src)
-    return testdir
+@dataclass
+class TestCase:
+    name: str  # name of the test case
+    existing_env_vars: Dict[str, str]  # preexisting environment variables
+    ini_contents: str  # pytest.ini contents
+    expected_env_vars: Dict[str, str]  # variables that are expected to be found after the plugin has done its job
 
+    # This helps pytest output to show test name for each test
+    def __str__(self):
+        return self.name
 
-def test_simple(example: pytest.Testdir) -> None:
-    (example.tmpdir / "pytest.ini").write_text("[pytest]\nenv = MAGIC=alpha", encoding="utf-8")
-    example.monkeypatch.setenv("_PATCH", "alpha")
-    result = example.runpytest()
+
```

Comment:
IMO, much more readable and maintainable, than creating all the boilerplate files, and boilerplate functions, for each of the test cases, especially when all of the test cases differ only by a few values, which is reflected in this parametrization.

For any test that might require more custom code, there is a test example left untouched.

What do you think?